### PR TITLE
Login error messages

### PR DESCRIPTION
**Pull request by lomamech**
Login error messages  …
- Update auth.js to catch No Permissions and No Project
- Add two lib errors for No Permission and No Project

Update test integration for recognized user without a project

### CODE REVIEW

File: server/controllers/auth.js [hunk 1]
```diff
@@ -52,12 +55,48 @@ function login(req, res, next) {
...
+    WHERE user.username = ? AND user.password = PASSWORD(?)`;
+
+  let sqlPermission = `
+    SELECT permission.id FROM permission
+    JOIN user ON user.id = permission.user_id
+    WHERE user.username = ? AND user.password = PASSWORD(?)`;  
+
+  transaction = db.transaction();
+
+  transaction
```
Comment (sfount) [inline server/controllers/auth.js]: MySQL transactions are very useful for when we need to roll back changes, however they are expensive to perform as they lock MySQL tables during execution. 

As all of these operations only involve reads I recommend that a transaction is not used. 

Instead this section could read like this: 

```js
db.exec(sqlTestUser) 
  .then((result) => { 
    // comment describing was result was
   if (result.length) { throw new BadRequest() }
   return db.exec(sqlTestProject);
  })
  .then // ...
```

File: server/lib/errors/NoPermission.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+const util = require('util');
+
+/**
+ * @class NoPermission
```
Comment (sfount) [inline server/lib/errors/NoPermission.js]: The `lib/BadRequest` class allows you to specify a key that will be sent to the user. In order to have less server code to maintain I suggest that these cases are handled as `BadRequest`s with unique error codes passed in the key field.

**Review by sfount — DISMISSED** [review]
The functionality of this pull request looks exactly correct! I have made some comments about small changes that would help the maintainability of the codebase.

**Comment by sfount** [comment]
I notice the `Unautharized` does not allow you to set a translation key for the user to see. This could be updated and used along with `BadRequest`

**Comment by jniles** [comment]
`Unauthorized` gained that functionality in an [unmerged PR](https://github.com/IMA-WorldHealth/bhima-2.X/pull/1133).  Should I merge that PR so it can be used here?

File: server/lib/errors/Unauthorized.js [hunk 3]
```diff
@@ -29,7 +29,9 @@ const util = require('util');
  *
  * @requires util
  */
-function Unauthorized(description) {
+
+function Unauthorized(description, key) {
+  'use strict';
```
Comment (jniles) [inline server/lib/errors/Unauthorized.js:34]: There is not need to add `use strict` anymore.  We are forcing the server to run in strict mode via an npm package.  It also shouldn't be necessary if you are running the latest NodeJS.

**Review by jniles — APPROVED** [review]
LGTM.  This functionality is important as users are getting confused when they are given generic error messages without any direction about what to do next.

:+1:

**Pull request merged**